--- original/Cli-38.java
+++ repaired/Cli-38.java
@@ -9,6 +9,10 @@
     // remove leading "-" and "=value"
     int pos = token.indexOf("=");
     String optName = pos == -1 ? token.substring(1) : token.substring(1, pos);
-    return options.hasShortOption(optName);
+    if (options.hasShortOption(optName))
+    {
+        return true;
+    }
     // check for several concatenated short options
+    return optName.length() > 0 && options.hasShortOption(String.valueOf(optName.charAt(0)));
 }
