--- original/Closure-97.java
+++ thinkrepair/Closure-97.java
@@ -41,10 +41,9 @@
         result = lvalInt >> rvalInt;
         break;
       case Token.URSH:
-        // JavaScript handles zero shifts on signed numbers differently than
-        // Java as an Java int can not represent the unsigned 32-bit number
-        // where JavaScript can so use a long here.
-        result = lvalInt >>> rvalInt;
+        // Use a long to promote lval to an unsigned 32-bit integer 
+        long unsignedLval = ((long)lvalInt) & 0xffffffffL;
+        result = unsignedLval >>> rvalInt;
         break;
       default:
         throw new AssertionError("Unknown shift operator: " +
