--- original/Compress-30.java
+++ repaired/Compress-30.java
@@ -8,16 +8,19 @@
     }
     if (offs + len > dest.length) {
         throw new IndexOutOfBoundsException("offs(" + offs + ") + len("
-                                            + len + ") > dest.length(" + dest.length + ").");
+                                            + len + ") > dest.length(" + dest.length + ")");
     }
     if (this.in == null) {
         throw new IOException("stream closed");
     }
+    if (len == 0) { // Added line to handle case len == 0
+        return 0;
+    }
 
-    final int hi = offs + len;
+    final int end = offs + len;
     int destOffs = offs;
     int b;
-    while (destOffs < hi && ((b = read0()) >= 0)) {
+    while (destOffs < end && ((b = read0()) >= 0)) {
         dest[destOffs++] = (byte) b;
         count(1);
     }
