--- original/JacksonXml-3.java
+++ repaired/JacksonXml-3.java
@@ -66,9 +66,8 @@
         }
         break;
     case XmlTokenStream.XML_ATTRIBUTE_VALUE:
-        _currText = _xmlTokens.getText();
         _currToken = JsonToken.VALUE_STRING;
-        break;
+        return (_currText = _xmlTokens.getText());
     case XmlTokenStream.XML_TEXT:
         _currText = _xmlTokens.getText();
         if (_mayBeLeaf) {
