--- original/Math-69.java
+++ thinkrepair/Math-69.java
@@ -9,7 +9,8 @@
             } else {
                 double r = correlationMatrix.getEntry(i, j);
                 double t = Math.abs(r * Math.sqrt((nObs - 2)/(1 - r * r)));
-                out[i][j] = 2 * (1 - tDistribution.cumulativeProbability(t));
+                double pValue = tDistribution.cumulativeProbability(-t) * 2;
+                out[i][j] = pValue;
             }
         }
     }
