--- original/JacksonDatabind-67.java
+++ repaired/JacksonDatabind-67.java
@@ -16,9 +16,10 @@
     // the only non-standard thing is this:
     if (deser == null) {
         if (type.isEnumType()) {
-            return _createEnumKeyDeserializer(ctxt, type);
+            deser = _createEnumKeyDeserializer(ctxt, type);
+        } else {
+            deser = StdKeyDeserializers.findStringBasedKeyDeserializer(config, type);
         }
-        deser = StdKeyDeserializers.findStringBasedKeyDeserializer(config, type);
     }
     // and then post-processing
     if (deser != null) {
