--- original/JacksonDatabind-24.java
+++ thinkrepair/JacksonDatabind-24.java
@@ -2,7 +2,10 @@
     if (_dateFormat == df) {
         return this;
     }
-    TimeZone tz = (df == null) ? _timeZone : df.getTimeZone();
+    TimeZone tz = (_timeZone != null) ? _timeZone : TimeZone.getDefault();
+    if (df != null) {
+        df.setTimeZone(tz);
+    }
     return new BaseSettings(_classIntrospector, _annotationIntrospector, _visibilityChecker, _propertyNamingStrategy, _typeFactory,
             _typeResolverBuilder, df, _handlerInstantiator, _locale,
             tz, _defaultBase64);
