--- original/JacksonDatabind-5.java
+++ repaired/JacksonDatabind-5.java
@@ -22,7 +22,12 @@
                  */
             } else {
                 // Well, or, as per [Issue#515], multi-level merge within mixins...
+                am = mixIns.find(m);
+                if (am != null) {
+                    _addMixUnders(m, am);
+                } else {
                     mixIns.add(_constructMethod(m));
+                }
             }
         }
     }
