--- original/JacksonDatabind-102.java
+++ repaired/JacksonDatabind-102.java
@@ -3,9 +3,6 @@
     {
         // Note! Should not skip if `property` null since that'd skip check
         // for config overrides, in case of root value
-        if (property == null) {
-            return this;
-        }
         JsonFormat.Value format = findFormatOverrides(serializers, property, handledType());
         if (format == null) {
             return this;
