--- original/Mockito-29.java
+++ thinkrepair/Mockito-29.java
@@ -1,7 +1,12 @@
 public void describeTo(Description description) {
     description.appendText("same(");
     appendQuoting(description);
-    description.appendText(wanted.toString());
+    if (wanted != null) {
+        description.appendText(wanted.toString());
+    }
+    else {
+        description.appendText("null");
+    }
     appendQuoting(description);
     description.appendText(")");
 }
