--- original/Jsoup-27.java
+++ repaired/Jsoup-27.java
@@ -3,8 +3,9 @@
     Matcher m = charsetPattern.matcher(contentType);
     if (m.find()) {
         String charset = m.group(1).trim();
+        if (Charset.isSupported(charset)) return charset;
         charset = charset.toUpperCase(Locale.ENGLISH);
-        return charset;
+        if (Charset.isSupported(charset)) return charset;
     }
     return null;
 }
