--- original/Lang-42.java
+++ repaired/Lang-42.java
@@ -1,15 +1,17 @@
 public void escape(Writer writer, String str) throws IOException {
-    int len = str.length();
+    final int len = str.length();
     for (int i = 0; i < len; i++) {
-        char c = str.charAt(i);
-        String entityName = this.entityName(c);
+        int codePoint = str.codePointAt(i);
+        i += Character.charCount(codePoint) - 1;
+
+        String entityName = this.entityName(codePoint);
         if (entityName == null) {
-            if (c > 0x7F) {
+            if (codePoint > 0x7F) {
                 writer.write("&#");
-                writer.write(Integer.toString(c, 10));
+                writer.write(Integer.toString(codePoint, 10));
                 writer.write(';');
             } else {
-                writer.write(c);
+                writer.write(codePoint);
             }
         } else {
             writer.write('&');
