--- original/JxPath-8.java
+++ repaired/JxPath-8.java
@@ -18,6 +18,12 @@
         return containsMatch((Iterator) right, left);
     }
     double ld = InfoSetUtil.doubleValue(left);
+    if (Double.isNaN(ld)) {
+        return false;
+    }
     double rd = InfoSetUtil.doubleValue(right);
+    if (Double.isNaN(rd)) {
+        return false;
+    }
     return evaluateCompare(ld == rd ? 0 : ld < rd ? -1 : 1);
 }
